data/mods/gen1/moves.ts

```diff
@@ -403,20 +412,41 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 			this.add('-clearallboost', '[silent]');
 			for (const pokemon of this.getAllActive()) {
 				pokemon.clearBoosts();
-
 				if (pokemon !== source) {
 					pokemon.cureStatus(true);
 				}
 				if (pokemon.status === 'tox') {
 					pokemon.setStatus('psn');
 				}
-				for (const id of Object.keys(pokemon.volatiles)) {
-					if (id === 'residualdmg') {
-						pokemon.volatiles[id].counter = 0;
-					} else {
-						pokemon.removeVolatile(id);
-						this.add('-end', pokemon, id, '[silent]');
-					}
+				// should only clear a specific set of volatiles and does not clear the toxic counter
+				const silentHack = '|[silent]';
+				if ('disable' in pokemon.volatiles) {
+					pokemon.removeVolatile('disable');
+					this.log[this.log.length - 1] += silentHack;
+				}
+				if ('confusion' in pokemon.volatiles) {
+					pokemon.removeVolatile('confusion');
+					this.log[this.log.length - 1] += silentHack;
+				}
+				if ('mist' in pokemon.volatiles) {
+					pokemon.removeVolatile('mist');
+					this.log[this.log.length - 1] += silentHack;
+				}
+				if ('focusenergy' in pokemon.volatiles) {
+					pokemon.removeVolatile('focusenergy');
+					this.log[this.log.length - 1] += silentHack;
+				}
+				if ('leechseed' in pokemon.volatiles) {
+					pokemon.removeVolatile('leechseed');
+					this.log[this.log.length - 1] += silentHack;
+				}
+				if ('lightscreen' in pokemon.volatiles) {
+					pokemon.removeVolatile('lightscreen');
+					this.log[this.log.length - 1] += silentHack;
+				}
+				if ('reflect' in pokemon.volatiles) {
+					pokemon.removeVolatile('reflect');
+					this.log[this.log.length - 1] += silentHack;
 				}
```

Comment:
Hey there, I didn't follow this exactly because I actually messed up here, and the first three need the silent hack whereas the last 4 are treated differently. Doing this based off of https://gist.github.com/scheibo/ddb13426a10335b7a471b92c5b258b49#file-haze-patch=

I did fix all of the silent hack ones to be done in a loop though.